src/status_im/chat/events.cljs

```diff
@@ -175,37 +154,36 @@
 (handlers/register-handler-fx
   :initialize-chats
   [(re-frame/inject-cofx :all-stored-chats)
+   (re-frame/inject-cofx :get-stored-messages)
    (re-frame/inject-cofx :stored-unviewed-messages)
-   (re-frame/inject-cofx :get-stored-unanswered-requests)
-   (re-frame/inject-cofx :get-last-stored-message)
-   (re-frame/inject-cofx :get-message-previews)]
+   (re-frame/inject-cofx :get-stored-unanswered-requests)]
   (fn [{:keys [db
                all-stored-chats
                stored-unanswered-requests
-               stored-unviewed-messages
-               get-last-stored-message
-               message-previews]} _]
-    (let [{:accounts/keys [account-creation?] :contacts/keys [contacts]} db
-          new-db (unviewed-messages-model/load-unviewed-messages db stored-unviewed-messages)
-          event  [:load-default-contacts!]]
+               get-stored-messages
+               stored-unviewed-messages]} _]
+    (let [{:accounts/keys [account-creation?] :contacts/keys [contacts]} db 
+          load-default-contacts-event [:load-default-contacts!]]
       (if account-creation?
-        {:db new-db
-         :dispatch event}
-        (let [chat->message-id->request (reduce (fn [acc {:keys [chat-id message-id] :as request}]
+        {:db db
+         :dispatch load-default-contacts-event}
+        (let [chat->unviewed-messages (unviewed-messages-model/index-unviewed-messages stored-unviewed-messages)
+              chat->message-id->request (reduce (fn [acc {:keys [chat-id message-id] :as request}]
                                                   (assoc-in acc [chat-id message-id] request))
                                                 {}
                                                 stored-unanswered-requests)
-              chats (->> all-stored-chats
-                         (map (fn [{:keys [chat-id] :as chat}]
-                                [chat-id (assoc chat
-                                                :last-message (get-last-stored-message chat-id)
-                                                :requests (get chat->message-id->request chat-id))]))
-                         (into {}))]
-          (-> new-db
-              (assoc-in [:message-data :preview] message-previews) 
+              chats (reduce (fn [acc {:keys [chat-id] :as chat}]
```

Comment:
This would be a perfect fit for an extra method: would help code reading and allow testing if needed.

Reply:
Will do

Reply:
Sure, but that function is ugly because of the one main reason - it has to take in account special console initalisation, that's why the `account-creation?` check and calling `init-console`.
Once we refactor the app startup sequence (which is very, very bad now, super fragile and requires special care of the console, which unfortunately can't be treated like any other chat, even if it should be that way), the ugliness will disappear and the function will actually be quite small and nice to read.